libdnf/dnf-repo.cpp

```diff
@@ -969,6 +969,8 @@ dnf_repo_set_keyfile_data(DnfRepo *repo, gboolean reloadFromGKeyFile, GError **e
         dnf_repo_apply_setopts(*conf, repoId);
     }
 
+    conf->basecachedir().set(libdnf::Option::Priority::REPOCONFIG, dnf_context_get_cache_dir(priv->context));
```

Comment:
Not sure if this is the right place to do so, but we need to set basecachedir at some point.